sale_partner_selectable_option/models/sale_order.py

```diff
@@ -3,12 +3,21 @@
 
 from lxml import etree
 
-from odoo import api, models
+from odoo import _, api, models
+from odoo.exceptions import UserError
 
 
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
+    @api.constrains("partner_id")
+    def _check_partner_id(self):
+        for order in self:
+            if not order.partner_id.sale_selectable:
```

Comment:
Mybe you can add a skip key context to avoid this restriction in some background process like as import data..

Reply:
Done @sergio-teruel